### PR TITLE
YaruPageItem: add more flexibility

### LINKED ISSUE #229: YaruPageItem: use Icon not IconData property

### PR DESCRIPTION
**Pull request by Feichtmeier**
Fixes #229

### CODE REVIEW

**Comment by jpnurmi** [comment]
Could we consider a builder for the icon, too? It would allow one to fetch colors from the theme, for example.

If we make it a builder, we could pass the selected state to make it possible to have just one builder?

```
Widget Function(BuildContext context, bool selected)
```

File: lib/src/pages/layouts/yaru_page_item.dart [hunk 1]
```diff
@@ -4,22 +4,16 @@ class YaruPageItem {
...
   /// We recommend to use [YaruPageItemTitle] here to avoid line wrap
   final WidgetBuilder titleBuilder;
 
   final WidgetBuilder builder;
-  final IconData iconData;
-  final Widget? itemWidget;
-  final Widget? selectedItemWidget;
-  final IconData? selectedIconData;
+  final Widget Function(BuildContext context, bool selected) icon;
   final bool Function(String value, BuildContext context)? searchMatches;
```
[the remark this answers is not in the document]
Comment (jpnurmi) [reply lib/src/pages/layouts/yaru_page_item.dart:17]: Yes, definitely! :+1:

**Comment by Feichtmeier** [comment]
If there is a convention for builders let's add it everywhere. WHat is the convention? WIth Builder suffix or without?

**Comment by Jupi007** [comment]
+1 for `iconBuilder`

**Comment by Feichtmeier** [comment]
((( very offtopic, but since we are gathered here in this beatiful virtual place... I think I discovered maybe the mistake that brought us to those page item title builders..... 

grafik

if the first row of the master detail page would be provided by the outer scaffold instead, then each page could have it's own apppage like it wants, without us needing a builder to bring it back into the tree .... :laughing: or am I wrong? wow this would be so funny xD )))

File: lib/src/pages/layouts/yaru_page_item.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (Feichtmeier) [reply lib/src/pages/layouts/yaru_page_item.dart]: nvm it is a small change, I removed it here :) Need to make some fixes in settings anyways after this is in

**Comment by jpnurmi** [comment]
> if the first row of the master detail page would be provided by the outer scaffold instead, then each page could have it's own ~apppage~ page like it wants with the appbar it wants, without us needing a builder to bring it back into the tree .... laughing or am I wrong? wow this would be so funny xD )))

You lost me there. :) Does the app know where the split is?

**Comment by Feichtmeier** [comment]
Well if you just use an AppBar yes. The height of the appbar is defined in yaru :shrug:

**Comment by Feichtmeier** [comment]
The only thing we would need to insert from outside is the appbar inside the master left panel then

grafik

**Comment by jpnurmi** [comment]
Right, so you just want to remove the automatic creation of the right-side appbar? :+1:

**Comment by Feichtmeier** [comment]
Hm maybe not in this release? In some other release
The removal of all those properties just opened my eyes xD 
We need the titles to be localized so a Map<Widget, Widget> with ListTiles mapped to Pages would not help so we need the title builders anyways :shrimp:

**Pull request merged**